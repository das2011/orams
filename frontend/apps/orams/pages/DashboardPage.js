import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter, Switch, Route } from 'react-router-dom'
import Dashboard from 'orams/components/Dashboard/Dashboard'

export class DashboardPage extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    const { match } = this.props

    return (
      <Switch>
        <Route exact path={match.url} render={() => <Dashboard {...this.props} />} />
      </Switch>
    )
  }
}

DashboardPage.propTypes = {
  match: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  errorMessage: state.app.errorMessage,
  successMessage: state.app.successMessage
})

export default withRouter(connect(mapStateToProps)(DashboardPage))
