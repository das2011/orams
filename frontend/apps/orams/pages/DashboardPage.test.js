import React from 'react'
import snapshotRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { DashboardPage } from './DashboardPage'

describe('DashboardPage component', () => {
  it('renders correctly', () => {
    const props = {
      errorMessage: null,
      successMessage: null,
      match: {}
    }
    const tree = snapshotRenderer
      .create(
        <MemoryRouter>
          <DashboardPage {...props} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
