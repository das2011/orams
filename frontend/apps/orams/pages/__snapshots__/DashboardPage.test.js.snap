// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage component renders correctly 1`] = `
<main>
  <div>
    DASHBOARD
  </div>
</main>
`;
